processing/src/main/java/org/apache/druid/segment/join/HashJoinSegmentStorageAdapter.java

```diff
@@ -52,6 +54,11 @@
   private final StorageAdapter baseAdapter;
   private final List<JoinableClause> clauses;
 
+  // A reference to the last JoinFilterSplit created during a makeCursors call,
+  // saved and exposed so that tests can verify the filter splitting behavior.
+  @VisibleForTesting
+  private JoinFilterAnalyzer.JoinFilterSplit previousJoinFilterSplitForTesting;
```

Comment:
Would you still need this if `JoinFilterAnalyzerTest` was created and had unit tests for `JoinFilterAnalyzer`?

Reply:
I restructured the tests in `JoinFilterAnalyzerTest` to call `JoinFilterAnalyzer.splitFilter` directly, and got rid of this testing hook. 

This means the filter split would be calculated twice in the tests (since I still call verifyCursors later), but I think that's fine for tests.